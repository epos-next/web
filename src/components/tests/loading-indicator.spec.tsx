import LoadingIndicator from "@components/loading-indicator";
import { render } from "@testing-library/react";
import React from "react";

it("should match snapshot", () => {
    const { asFragment } = render(<LoadingIndicator/>)
    expect(asFragment()).toMatchSnapshot()
})

it("should use radius", () => {
    const { asFragment } = render(<LoadingIndicator radius={2000}/>)
    expect(asFragment()).toMatchSnapshot()
})

it("should use color", () => {
    const { asFragment } = render(<LoadingIndicator color="#ccc123"/>)
    expect(asFragment()).toMatchSnapshot()
});
