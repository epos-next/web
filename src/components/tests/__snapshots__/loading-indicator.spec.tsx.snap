// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match snapshot 1`] = `
<DocumentFragment>
  .c0 .dot-stretching {
  position: relative;
  width: 10px;
  height: 10px;
  border-radius: 5px;
  background-color: var(--disabled);
  color: var(--disabled);
  -webkit-transform: scale(1.25,1.25);
  -ms-transform: scale(1.25,1.25);
  transform: scale(1.25,1.25);
  -webkit-animation: dotStretching 2s infinite ease-in;
  animation: dotStretching 2s infinite ease-in;
}

.c0 .dot-stretching::before,
.c0 .dot-stretching::after {
  content: '';
  display: inline-block;
  position: absolute;
  top: 0;
}

.c0 .dot-stretching::before {
  width: 10px;
  height: 10px;
  border-radius: 50%;
  background-color: var(--disabled);
  color: var(--disabled);
  -webkit-animation: dotStretchingBefore 2s infinite ease-in;
  animation: dotStretchingBefore 2s infinite ease-in;
}

.c0 .dot-stretching::after {
  width: 10px;
  height: 10px;
  border-radius: 50%;
  background-color: var(--disabled);
  color: var(--disabled);
  -webkit-animation: dotStretchingAfter 2s infinite ease-in;
  animation: dotStretchingAfter 2s infinite ease-in;
}

<div
    class="c0"
    color="var(--disabled)"
    radius="10"
  >
    <div
      class="dot-stretching"
    />
  </div>
</DocumentFragment>
`;

exports[`should use color 1`] = `
<DocumentFragment>
  .c0 .dot-stretching {
  position: relative;
  width: 10px;
  height: 10px;
  border-radius: 5px;
  background-color: #ccc123;
  color: #ccc123;
  -webkit-transform: scale(1.25,1.25);
  -ms-transform: scale(1.25,1.25);
  transform: scale(1.25,1.25);
  -webkit-animation: dotStretching 2s infinite ease-in;
  animation: dotStretching 2s infinite ease-in;
}

.c0 .dot-stretching::before,
.c0 .dot-stretching::after {
  content: '';
  display: inline-block;
  position: absolute;
  top: 0;
}

.c0 .dot-stretching::before {
  width: 10px;
  height: 10px;
  border-radius: 50%;
  background-color: #ccc123;
  color: #ccc123;
  -webkit-animation: dotStretchingBefore 2s infinite ease-in;
  animation: dotStretchingBefore 2s infinite ease-in;
}

.c0 .dot-stretching::after {
  width: 10px;
  height: 10px;
  border-radius: 50%;
  background-color: #ccc123;
  color: #ccc123;
  -webkit-animation: dotStretchingAfter 2s infinite ease-in;
  animation: dotStretchingAfter 2s infinite ease-in;
}

<div
    class="c0"
    color="#ccc123"
    radius="10"
  >
    <div
      class="dot-stretching"
    />
  </div>
</DocumentFragment>
`;

exports[`should use radius 1`] = `
<DocumentFragment>
  .c0 .dot-stretching {
  position: relative;
  width: 2000px;
  height: 2000px;
  border-radius: 5px;
  background-color: var(--disabled);
  color: var(--disabled);
  -webkit-transform: scale(1.25,1.25);
  -ms-transform: scale(1.25,1.25);
  transform: scale(1.25,1.25);
  -webkit-animation: dotStretching 2s infinite ease-in;
  animation: dotStretching 2s infinite ease-in;
}

.c0 .dot-stretching::before,
.c0 .dot-stretching::after {
  content: '';
  display: inline-block;
  position: absolute;
  top: 0;
}

.c0 .dot-stretching::before {
  width: 2000px;
  height: 2000px;
  border-radius: 50%;
  background-color: var(--disabled);
  color: var(--disabled);
  -webkit-animation: dotStretchingBefore 2s infinite ease-in;
  animation: dotStretchingBefore 2s infinite ease-in;
}

.c0 .dot-stretching::after {
  width: 2000px;
  height: 2000px;
  border-radius: 50%;
  background-color: var(--disabled);
  color: var(--disabled);
  -webkit-animation: dotStretchingAfter 2s infinite ease-in;
  animation: dotStretchingAfter 2s infinite ease-in;
}

<div
    class="c0"
    color="var(--disabled)"
    radius="2000"
  >
    <div
      class="dot-stretching"
    />
  </div>
</DocumentFragment>
`;
